### PR TITLE
Bubble Sort Project for ruby curriculum

### PR DESCRIPTION
**Pull request by carlosprietobarron**
This is the first project for the Ruby curriculum

### CODE REVIEW

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, but there are a few more things required. 
- [ ] `bubble_sort `
`p bubble_sort([3,4,52,31,7,5])` should return `[3,4,5,7,31,52]`
- [ ] `bubble_sort_by method` is not implemented ,lets fix that

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by carlosprietobarron** [comment]
Thank you very much for your review, it was really helpful to improve our performance.
We have made the changes requested.
Thanks

File: sort.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  i = 0
+  while i < loops - 1
+    j = 0
+    while j < loops - 1
+      arr[j], arr[j + 1] = arr[j + 1], arr[j] if arr[j] > arr[j + 1]
+      j += 1
+    end
+    i += 1
+  end
+  puts arr
```
Comment (shshamim63) [inline sort.rb]: The task of the sort is to return a sorted array. But I can see puts. Please remove `puts` and it should be only `arr`.

File: sort.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  (arr.length - 1).step(1, -1) do |i|
+    swapped = false
+    (0...i).each do |j|
+      swapped = true if yield(arr[j], arr[j + 1]).to_i.positive?
+      arr[j], arr[j + 1] = arr[j + 1], arr[j] if yield(arr[j], arr[j + 1]).to_i.positive?
+    end
+    break unless swapped
+  end
+
+  puts arr
```
Comment (shshamim63) [inline sort.rb]: Same as above. It does not require puts.

**Review by shshamim63 — APPROVED** [review]
## Status :tada: Approved :tada: 
___
Hi @carlosprietobarron and @sebGilR :open_hands:,
You guys are almost there. I just found two simple cases, though those are mandatory but not a very tough job to resolve. Sort returns a sorted array, that is it. Really simple job. But you are providing output on the console. That is against the basic functionality of a sort method. So check my comments below and resolve the issues. As they are not a tough job, I am approving this in advance, but keeping an :eye:. :rofl: 
___
#### This :+1:  is from me. you can go ahead.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Twitter](https://twitter.com/Shshamim090)

**Pull request merged**